src/log-store/src/kafka/log_store.rs

```diff
@@ -104,3 +240,45 @@ impl LogStore for KafkaLogStore {
         Ok(())
     }
 }
+
+/// Tries to get the physical offset of the entry with the given entry id.
+// TODO(niebayes): a mapping between entry id and entry offset needs to maintained at somewhere.
+// One solution is to store the mapping at a specific Kafka topic. Each time the mapping is updated,
+// a new record is constructed and appended to the topic. On initializing the log store, the latest
+// record is pulled from Kafka cluster.
+//
+// The second solution is to store the mapping at the kv backend. We design a dedicated key, e.g. ID_TO_OFFSET_MAP_KEY.
+// Each time the mapping is updated, the map is serialized into a vector of bytes and stored into the kv backend at the given key.
+// On initializing the log store, the map is deserialized from the kv backend.
+//
+// The third solution is to store the offset for each region separately. More specifically, when flushed, the
```

Comment:
I would like the third way. The offset is a region metadata.

Reply:
Although the third way is the easiest way to implement, maintaining a Kafka offset in the region's metadata brings many Kafka-specific logics into the storage engine and may confuse people.

It means that the Kafka log store can work properly without a third-party manifest store. I think we need to further discuss this problem.

Reply:
cc @MichaelScofield

Reply:
After discussions, the agreed solution is to refactor the usage of entry id and don't use entry offset instead.